src/core/mac/mac_frame.hpp

```diff
@@ -1043,328 +1043,365 @@ class Frame : public otRadioFrame
+    void SetRssi(int8_t aRssi) { mInfo.mRxInfo.mRssi = aRssi; }
 
     /**
-     * This const method returns a pointer to the MAC Header.
+     * This method returns the receive Link Quality Indicator.
      *
-     * @returns A pointer to the MAC Header.
+     * @returns The receive Link Quality Indicator.
      *
      */
-    const uint8_t *GetHeader(void) const { return GetPsdu(); }
+    uint8_t GetLqi(void) const { return mInfo.mRxInfo.mLqi; }
 
     /**
-     * This method returns a pointer to the MAC Payload.
+     * This method sets the receive Link Quality Indicator.
      *
-     * @returns A pointer to the MAC Payload.
+     * @param[in]  aLqi  The receive Link Quality Indicator.
      *
      */
-    uint8_t *GetPayload(void) { return const_cast<uint8_t *>(const_cast<const Frame *>(this)->GetPayload()); }
+    void SetLqi(uint8_t aLqi) { mInfo.mRxInfo.mLqi = aLqi; }
 
     /**
-     * This const method returns a pointer to the MAC Payload.
+     * This method indicates whether or not the received frame is acknowledged with frame pending set.
      *
-     * @returns A pointer to the MAC Payload.
+     * @retval TRUE   This frame is acknowledged with frame pending set.
+     * @retval FALSE  This frame is acknowledged with frame pending not set.
      *
      */
-    const uint8_t *GetPayload(void) const;
+    bool IsAckedWithFramePending(void) const { return mInfo.mRxInfo.mAckedWithFramePending; }
 
     /**
-     * This method returns a pointer to the MAC Footer.
+     * This method returns the timestamp when the frame was received.
      *
-     * @returns A pointer to the MAC Footer.
+     * @returns The timestamp when the frame was received, in microseconds.
      *
      */
-    uint8_t *GetFooter(void) { return const_cast<uint8_t *>(const_cast<const Frame *>(this)->GetFooter()); }
+    const uint64_t &GetTimestamp(void) const { return mInfo.mRxInfo.mTimestamp; }
 
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
     /**
-     * This const method returns a pointer to the MAC Footer.
+     * This method gets the offset to network time.
      *
-     * @returns A pointer to the MAC Footer.
+     * @returns  The offset to network time.
      *
      */
-    const uint8_t *GetFooter(void) const;
+    int64_t ComputeNetworkTimeOffset(void) const
+    {
+        return static_cast<int64_t>(GetTimeIe()->GetTime() - GetTimestamp());
+    }
 
     /**
-     * This method returns the timestamp when the frame was received.
+     * This method gets the time sync sequence.
      *
-     * @returns The timestamp when the frame was received, in microseconds.
+     * @returns  The time sync sequence.
      *
      */
-    const uint64_t &GetTimestamp(void) const { return mInfo.mRxInfo.mTimestamp; }
+    uint8_t ReadTimeSyncSeq(void) const { return GetTimeIe()->GetSequence(); }
+#endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+};
 
-#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+/**
+ * This class implements IEEE 802.15.4 MAC tx frame generation and parsing.
```

Comment:
how about replace with: "This class supports IEEE 802.15.4 MAC frame generation for transmission."